app/Auth0/Auth0ClientsForIntegration.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Auth0;
+
+use InvalidArgumentException;
+use Ramsey\Uuid\UuidInterface;
+
+final class Auth0ClientsForIntegration
+{
+    private array $auth0ClientPerTenant = [];
+
+    public function __construct(
+        public readonly UuidInterface $integrationId,
+        Auth0Client ...$auth0Clients
+    ) {
+        foreach ($auth0Clients as $auth0Client) {
+            if (!$auth0Client->integrationId->equals($this->integrationId)) {
+                throw new InvalidArgumentException(
+                    sprintf(
+                        'Client with id %s of integration id %s does not belong to expected integration id %s',
+                        $auth0Client->clientId,
+                        $auth0Client->integrationId->toString(),
+                        $this->integrationId->toString()
+                    )
+                );
+            }
+            $this->auth0ClientPerTenant[$auth0Client->tenant->value] = $auth0Client;
+        }
+    }
+
+    public function getClientForTenant(Auth0Tenant $tenant): ?Auth0Client
```

Comment:
This method doesn't seem to be used.

Reply:
Ah true, but we will need it later (for example in the frontend to show only the credentials for the test tenant for non-activated integrations) and it seems simple enough to keep?